storage/scylladb_Permanode.c

```diff
@@ -0,0 +1,614 @@
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+  result = cass_future_get_result(future);
+  iterator = cass_iterator_from_result(result);
+
+  while (cass_iterator_next(iterator)) {
+    const CassRow* row = cass_iterator_get_row(iterator);
+    size_t len;
+    const cass_byte_t* buf;
+
+    if (cass_value_get_bytes(cass_row_get_column_by_name(row, column_name), &buf, &len) != CASS_OK) {
+      ta_log_error("Get column failed\n");
+      ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+      goto end_iterate;
+    }
+    if (hash243_queue_push(res_queue, (flex_trit_t*)buf) != RC_OK) {
+      ta_log_error("Fail to push buf into queue\n");
+      ret = SC_STORAGE_OOM;
+      goto end_iterate;
+    }
+    ret = SC_OK;
+  }
+
+end_iterate:
+  cass_result_free(result);
+  cass_iterator_free(iterator);
+
+exit:
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return ret;
+}
+
+static status_t get_trytes(CassSession* session, CassStatement* statement, hash8019_queue_t* res_queue) {
+  status_t ret = SC_CCLIENT_NOT_FOUND;
+  CassFuture* future = NULL;
+  const CassResult* result;
+  CassIterator* iterator;
+
+  future = cass_session_execute(session, statement);
+  cass_future_wait(future);
+
+  if (cass_future_error_code(future) != CASS_OK) {
+    print_error(future);
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+  result = cass_future_get_result(future);
+  iterator = cass_iterator_from_result(result);
+
+  if (cass_iterator_next(iterator)) {
+    const CassRow* row = cass_iterator_get_row(iterator);
+    size_t len;
+    const cass_byte_t* buf;
+    if (cass_value_get_bytes(cass_row_get_column_by_name(row, "trytes"), &buf, &len) != CASS_OK) {
+      ta_log_error("Fail to get column : trytes\n");
+      ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+      goto end_iterate;
+    }
+    if (hash8019_queue_push(res_queue, (flex_trit_t const* const)buf) != RC_OK) {
+      ta_log_error("%s\n", "SC_STORAGE_OOM");
+      ret = SC_STORAGE_OOM;
+    }
+    ret = SC_OK;
+  }
+
+end_iterate:
+  cass_result_free(result);
+  cass_iterator_free(iterator);
+
+exit:
+  cass_future_free(future);
+  cass_statement_free(statement);
+  return ret;
+}
+
+static status_t get_inclustion_status_from_transaction_table(CassSession* session, get_inclusion_states_res_t* res,
+                                                             const flex_trit_t* hash) {
+  status_t ret = SC_OK;
+  CassStatement* statement = NULL;
+  const CassPrepared* prepared = NULL;
+
+  char* query = "SELECT COUNT(*) FROM transaction WHERE hash = ? LIMIT 1";
+  if (prepare_query(session, query, &prepared) != CASS_OK) {
+    ta_log_error("%s\n", "Prepare SELECT query failed");
+    return SC_STORAGE_CASSANDRA_QUREY_FAIL;
+  }
+
+  statement = cass_prepared_bind(prepared);
+  cass_statement_bind_bytes_by_name(statement, "hash", (cass_byte_t*)hash, NUM_TRYTES_HASH);
+  ret = get_count(session, statement, res);
+  cass_prepared_free(prepared);
```

Comment:
The `statement` also need to be freed, no?
Same for the below functions.

Reply:
It will be freed in `get_count`.